### ISSUE TITLE
gameplay is not saved!

### ISSUE THREAD

**Issue by tach-tm** [TrinityCore/TrinityCore#27448]
### Description

you play but after reboot starts from the starting point!
gameplay is not saved! 
complete reinstallation of the database did not give any results.

### Expected behaviour

Should be saved in the place where you left the game! and after restarting the server and entering the game, it should start from the place where you left the game last time!

### Steps to reproduce the problem

just go into the game and play!

### Branch

master

### TC rev. hash/commit

TrinityCore rev. fe489e2be131 2021-12-21 01:20:51 +0100 (master branch) (Win64, RelWithDebInfo, Static) (worldserver-daemon) ready...

### Operating system

Windows  11

### Custom changes

there were only changes in worldserver.conf SERVER RATES

**Comment by MaxtorCoder** [comment]
Okay so 1 question. Do you just close the server..? Or did you logout first. Because you need to log out first or shutdown the server with the `.server shutdown` command...

**Comment by tach-tm** [comment]
server shutdown occurs CTRL + C, this started recently with the last 2 or 4 update!

**Comment by MaxtorCoder** [comment]
I am not asking that, do you force close the server or with the shutdown command (or just logout on the character and then close)

**Comment by tach-tm** [comment]
Yes, I close it myself! after I play! or I will test it! or let my daughter play! but in any case, after the reboot, it will start from the initial state!

**Comment by MaxtorCoder** [comment]
Did you logout first..? Because force closing the application (ALT + F4 or the close button) won't save anything as it does not call any saving methods. Logging out of the character or using the shutdown command does.

**Comment by tach-tm** [comment]
Yes, I close it myself! after I play! or I will test it! or let my daughter play! but in any case, after the reboot, it will start from the initial state!

**Comment by Shauren** [comment]
Cannot reproduce, character still got saved even if I was online when pressing ctrl+c

**Comment by tach-tm** [comment]
I can’t define it, but character remains unchanged, but at the same time, some changes still remain. Brought the error logs there! I think after this error this was not the case before.
```
SQL (p): INSERT INTO character_aura (guid, casterGuid, itemGuid, spell, effectMask, recalculateMask, difficulty, stackCount, maxDuration, remainTime, remainCharges, castItemId, castItemLevel) VALUES (2, BINARY, BINARY, 261764, 1, 1, 1, , 1, -1, -1, 0, 0, -1)
  [ERROR]: [1062] Duplicate entry '2- \ x02 \ x00 \ x00 \ x00 \ x00 \ x00 \ x00 \ x00 \ x00 \ x00 \ x00 \ x00 \ x00 \ x04 \ x00 \ x' for key 'PRIMARY'
```

and here is the end of the server log

```
MoveSplineInitArgs::Validate: expression '_checkPathLengths()' failed for Creature-0-1-0-0-42218-0000000569 Entry: 42218
MoveSplineInitArgs::Validate: expression '_checkPathLengths()' failed for Creature-0-1-0-0-68-0000000563 Entry: 68
MoveSplineInitArgs::Validate: expression '_checkPathLengths()' failed for Creature-0-1-0-0-721-000000050C Entry: 721
MoveSplineInitArgs::Validate: expression '_checkPathLengths()' failed for Creature-0-1-0-0-3937-00000004AE Entry: 3937
Received not handled opcode [CMSG_DISCARDED_TIME_SYNC_ACKS 0x3A3E (14910)] from [Player: Damir Player-1-00000002, Account: 1]
Received not handled opcode [CMSG_DISCARDED_TIME_SYNC_ACKS 0x3A3E (14910)] from [Player: Damir Player-1-00000002, Account: 1]
Received not handled opcode [CMSG_MOVE_INIT_ACTIVE_MOVER_COMPLETE 0x3A43 (14915)] from [Player: Damir Player-1-00000002, Account: 1]
Received not handled opcode [CMSG_QUERY_COUNTDOWN_TIMER 0x31AA (12714)] from [Player: Damir Player-1-00000002, Account: 1]
Received not handled opcode [CMSG_REQUEST_CONQUEST_FORMULA_CONSTANTS 0x32C7 (12999)] from [Player: Damir Player-1-00000002, Account: 1]
Received not handled opcode [CMSG_LFG_LIST_GET_STATUS 0x360B (13835)] from [Player: Damir Player-1-00000002, Account: 1]
Received not handled opcode [CMSG_REQUEST_LFG_LIST_BLACKLIST 0x32B7 (12983)] from [Player: Damir Player-1-00000002, Account: 1]
Received not handled opcode [CMSG_GET_LANDING_PAGE_SHIPMENTS 0x3310 (13072)] from [Player: Damir Player-1-00000002, Account: 1]
Halting process...
Prevented sending of [SMSG_AURA_UPDATE 0x2C22 (11298)] to non existent socket 1 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_AURA_UPDATE 0x2C22 (11298)] to non existent socket 1 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_AURA_UPDATE 0x2C22 (11298)] to non existent socket 1 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_SET_PCT_SPELL_MODIFIER 0x2C37 (11319)] to non existent socket 1 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_SET_PCT_SPELL_MODIFIER 0x2C37 (11319)] to non existent socket 1 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_SET_PCT_SPELL_MODIFIER 0x2C37 (11319)] to non existent socket 1 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_AURA_UPDATE 0x2C22 (11298)] to non existent socket 1 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_AURA_UPDATE 0x2C22 (11298)] to non existent socket 1 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_AURA_UPDATE 0x2C22 (11298)] to non existent socket 1 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_AURA_UPDATE 0x2C22 (11298)] to non existent socket 1 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_AURA_UPDATE 0x2C22 (11298)] to non existent socket 1 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_AURA_UPDATE 0x2C22 (11298)] to non existent socket 1 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_AURA_UPDATE 0x2C22 (11298)] to non existent socket 1 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_CANCEL_COMBAT 0x2945 (10565)] to non existent socket 0 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_CANCEL_COMBAT 0x2945 (10565)] to non existent socket 0 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_UPDATE_OBJECT 0x27B7 (10167)] to non existent socket 1 to [Player: Damir Player-1-00000002, Account: 1]
Prevented sending of [SMSG_LOGOUT_COMPLETE 0x2679 (9849)] to non existent socket 0 to [Player: Account: 1]
```

**Comment by funjoker** [comment]
Just some little point: don’t use so many exclamation marks pls. It feels rude.

**Comment by MaxtorCoder** [comment]
> Just some little point: don’t use so many exclamation marks pls. It feels rude.

More like annoying lol

**Comment by tach-tm** [comment]
Sorry if you feel rude and for mistakes. (I don’t know English so well I crawl as a google translator). Completely changed the base, it seems like it was saved and then stopped again. no idea why. I noticed that this error starts and then hangs.
```
SQL(p): INSERT INTO character_aura (guid, casterGuid, itemGuid, spell, effectMask, recalculateMask, difficulty, stackCount, maxDuration, remainTime, remainCharges, castItemId, castItemLevel) VALUES (1, BINARY, BINARY, 261764, 1, 1, 0, 1, -1, -1, 0, 0, -1)
 [ERROR]: [1062] Duplicate entry '1-\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x04\x00\x' for key 'PRIMARY'
```

**Comment by tach-tm** [comment]
found the reason. and the decision is wrong. I just removed the 'PRIMARY' key from the two tables "character_aura" and "character_aura_effect". all keys. and it began to persist. I just don’t know how to do it correctly.

**Comment by shamage** [comment]
According to your first diary, you published it looks similar to what I had before. For me, this caused some values in the DB to be overwritten incorrectly during the update. This caused a complete collapse of my server later. I suggest if you have a correct DB backup then use it for recovery. It will save you a lot of time to find bugs in the servers.
The second comment that you mention is that you are trying to insert duplicate data. I suggest using the command before that line

`DELETE FROM character_aura WHERE guid IN (1);`

```
SQL (p): INSERT INTO character_aura (guid, casterGuid, itemGuid, spell, effectMask, recalculateMask, difficulty, stackCount, maxDuration, remainTime, remainCharges, castItemId, castItemLevel) VALUES (1, BINARY, BINARY, 261764, 1, 1, 1 , 1, -1, -1, 0, 0, -1)
  [ERROR]: [1062] Duplicate entry '1- \ x01 \ x00 \ x00 \ x00 \ x00 \ x00 \ x00 \ x00 \ x00 \ x00 \ x00 \ x00 \ x00 \ x04 \ x00 \ x' for key 'PRIMARY'
```

**Comment by shamage** [comment]
Another piece of advice:
I didn't understand exactly how you quit the game and the server itself, but I recommend that you don't quit the game or the server in `ALT + F4` or by closing the window. Otherwise, something may become corrupted in the DB, and functional problems may occur after starting the server later, which are difficult to fix. Always quit `ALT + C `or with the command directly in the world ==>` .server shutdown` window

**Comment by tach-tm** [comment]
even when I play in the character base table, changes must go through (level increase and map coordinate changes), which persist after a certain time (even after the command .save) and so these changes do not go through. as if it freezes, and this is for all accounts. but after unbinding, the 'PRIMARY' key started to change the table. (during the game)

**Comment by tach-tm** [comment]
I cleaned them up at the expense of the tables "character_aura" and "character_aura_effect". voshe they were empty, but there the values began to be created and in the 2nd and 3rd columns some incomprehensible values are shown with 3 cubes when viewed in edit mode, it shows 1 cube. not readable. (I'll try to make a screen in the evening the table itself)

**Comment by tach-tm** [comment]
https://ltdfoto.ru/images/character_aura.png
https://ltdfoto.ru/images/character_aura_effect.png

**Comment by Shauren** [comment]
Do you have AllowTrackBothResources enabled in config?

**Comment by tach-tm** [comment]
Yes. thanks I got it. The AllowTrackBothResources parameter was enabled and it seems to be causing the character to freeze. I'll take a look. I can write all the errors that I find. (we just play with three (me, son and daughter for fun) and notice some of them although I did not play the original game, for example, my daughter tells every time where the fishing is (it doesn’t work). and my son needs to go beyond (I think he needs to teleport).
